pkg/dependencies/deps.go

```diff
@@ -0,0 +1,88 @@
+package dependencies
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"code-intelligence.com/cifuzz/pkg/log"
+	"code-intelligence.com/cifuzz/pkg/runfiles"
+)
+
+type Key string
+
+const (
+	CLANG           Key = "clang"
+	CMAKE               = "cmake"
+	LLVM_COV            = "llvm-cov"
+	LLVM_SYMBOLIZER     = "llvm-symbolizer"
+	LLVM_PROFDATA       = "llvm-profdata"
+)
+
+// Dependency represents a single dependency
+type Dependency struct {
+	finder runfiles.RunfilesFinder
+
+	Key        Key
+	MinVersion semver.Version
+	// these fields are used to implement custom logic to
+	// retrieve version or installation information for the
+	// specific dependency
+	GetVersion func(*Dependency) (*semver.Version, error)
+	Installed  func(*Dependency) bool
+}
+
+// Compares MinVersion against GetVersion
+func (dep *Dependency) CheckVersion() bool {
+	currentVersion, err := dep.GetVersion(dep)
+	if err != nil {
+		log.Errorf(err, "Unable to get current version for %s", dep.Key)
+		// we want to be lenient if we were not able to extract the version
+		return true
+	}
+
+	if currentVersion.Compare(&dep.MinVersion) == -1 {
+		log.Warnf("Dependency %s has invalid version %s (min. %s)", dep.Key, currentVersion.String(), dep.MinVersion.String())
+		return false
+	}
+	return true
+}
+
+func (dep *Dependency) Ok() bool {
+	if !dep.Installed(dep) {
+		log.Warnf("Dependency %s is not installed", dep.Key)
```

Comment:
I would prefer:
```suggestion
		log.Warnf("cifuzz requires %s, but it is not installed", dep.Key)
```

We could even pass in the command name as an argument and then print `cifuzz foo requires ...` instead.